uaclient/jobs/gcp_auto_attach.py

```diff
@@ -2,15 +2,25 @@
 Try to auto-attach in a GCP instance. This should only work
 if the instance has a new UA license attached to it
 """
+import base64
+import json
 import logging
 
 from uaclient import config, exceptions
 from uaclient.cli import action_auto_attach
+from uaclient.clouds.gcp import GCP_LICENSES, UAAutoAttachGCPInstance
 from uaclient.clouds.identity import get_cloud_type
+from uaclient.util import get_platform_info
 
 LOG = logging.getLogger(__name__)
 
 
+def get_licenses_from_identity(identityToken):
+    identity = base64.urlsafe_b64decode(identityToken.split(".")[1] + b"===")
+    identity = json.loads(identity)
+    return identity["google"]["compute_engine"]["license_id"]
```

Comment:
It seems we aren't doing any error handling if the IMDS data is unavailable (due to IMDS being unresponsive/timeout) or if the format of the identityToken doesn't meet our expectations with either b64decode, json.loads,  or our expectation that the keypath ["google"]["compute_engine"]["license_id"] exists. I think we probably want to better handle those types of unexpected errors right? Maybe we add a silent "pass" in these failure cases since we expect this data to be correct some point, but we don't want to fill the logs every 5 minutes with the same error (and we don't want to talk to the contract server in those cases where the format has changed).